selenium-tests/tests/contacts-test.js

```diff
@@ -0,0 +1,63 @@
+const { By, Key, Builder, until } = require("selenium-webdriver");
+const assert = require("assert");
+const chrome = require("selenium-webdriver/chrome");
+let driver;
+const chromeOptions = new chrome.Options();
+
+chromeOptions.addArguments("--headless");
+
+async function testContactsPage() {
+  try {
+    driver = await new Builder()
+      .forBrowser("chrome")
+      .setChromeOptions(chromeOptions)
+      .build();
+
+    await driver.get("http://localhost/contacts.html");
+
+    let alertingHeader = await driver.findElement(By.css(".myOrg-heading")).getText();
+    assert.equal(alertingHeader, "Alerting", "Alerting header text is not correct");
+
+    let addContactButton = await driver.findElement(By.id("new-contact-point"));
+    await addContactButton.click();
+
+    let contactFormDisplayed = await driver.findElement(By.id("contact-form-container")).isDisplayed();
+    assert.equal(contactFormDisplayed, true, 'Contact form container is not displayed after clicking "Add Contact Point"');
+
+    // Additional Tests for the Contact Form
+    let contactNameInput = await driver.findElement(By.id("contact-name"));
+    await contactNameInput.sendKeys("Test Contact Name");
+    let contactNameValue = await contactNameInput.getAttribute('value');
+    assert.equal(contactNameValue, "Test Contact Name", "Contact name input value is not correct");
+
+    // Test Dropdown for Contact Types
+    let contactTypeDropdown = await driver.findElement(By.id("contact-types"));
+    await contactTypeDropdown.click();
+    let slackOption = await driver.findElement(By.id("option-0"));
+    await slackOption.click();
+
+    // Test Slack Channel ID Input
+    let slackChannelInput = await driver.findElement(By.id("slack-channel-id"));
+    await slackChannelInput.sendKeys("123456");
+    let slackChannelValue = await slackChannelInput.getAttribute('value');
+    assert.equal(slackChannelValue, "123456", "Slack channel ID input value is not correct");
+
+    // Test Slack Token Input
+    let slackTokenInput = await driver.findElement(By.id("slack-token"));
+    await slackTokenInput.sendKeys("Token123");
+    let slackTokenValue = await slackTokenInput.getAttribute('value');
+    assert.equal(slackTokenValue, "Token123", "Slack token input value is not correct");
+
+    // Test Save Button Functionality
+    let saveContactButton = await driver.findElement(By.id("save-contact-btn"));
+    await saveContactButton.click();
+
```

Comment:
After save, add test to check if the contact was added in the table

Reply:
I am nearing the completion of testing for both HTML pages. However, I am encountering an error related to a read-only database when attempting to create a contact point and an alert rule on the 'Contacts' and 'All-Alerts' pages. Could you grant me the necessary permissions to create data, so that I can finalize the testing process?

Reply:
For what database? I think you should have any required databases on your local machine, so you wouldn't need us to give you permissions, but I might be wrong about that

Reply:
I wanted to update you regarding the earlier issue. It appears there was a minor technical glitch that temporarily prevented me from accessing the database. However, I am pleased to inform you that I have now completed the testing. Additionally, I have pushed the latest changes to the same pull request for your review and further action.